### PR TITLE
Refactor clrypt: remove M2Crypto dependency

### PR DESCRIPTION
**Pull request by zacharyvoase**
This is a significant two-part refactor:

1) I separated the bulk of clrypt's behavior into `EncryptedDirectory` and `OpenSSLKeypair` classes; this means we can test encryption and file management independently. `clrypt/__init__.py` simply composes a keypair constructed from the appropriate environment variables with an encrypted directory.

2) Instead of M2Crypto we use the OpenSSL command-line tool. It's pretty much guaranteed to be available everywhere, all of our uses of clrypt seem to be outside of the request flow so performance isn't much of a concern, and we will no longer have to deal with SWIG or OpenSSL headers.

### CODE REVIEW

**Comment by zacharyvoase** [comment]
screenshot 2015-12-14 20 07 44

also new

**Comment by zacharyvoase** [comment]
better
screenshot 2015-12-15 11 55 50

File: clrypt/__init__.py [hunk 1]
```diff
@@ -1,95 +1,63 @@
...
 
-    outpath = _encrypted_file_path(group, name, ext)
-    dirname = os.path.dirname(outpath)
-    if not os.path.isdir(dirname):
-        os.makedirs(dirname)
 
-    with open(outpath, 'w') as out_fp:
-        out = BIO.File(out_fp)
-        s.write(out, p7)
+def _find_encrypted_directory(current_dir, dirname='encrypted', limit=100):
```
Comment (bhatman) [inline clrypt/__init__.py]: minor, but consider using a for loop that terminates when you reach `/`?

File: clrypt/encdir.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    def encrypted_file_path(self, group, name, ext='yaml'):
+        """Get the path of an encrypted file with the given group and name."""
+        file_name = '%s-%s.%s.smime' % (self.keypair.get_key_id(), name, ext)
+        return os.path.join(self.encrypted_dir, group, file_name)
+
+    def read_file(self, group, name, ext='yaml'):
+        """Read the named file as a bytestring of decrypted plaintext."""
+        with open(self.encrypted_file_path(group, name, ext=ext)) as encrypted:
+            bytes = encrypted.read()
```
Comment (bhatman) [inline clrypt/encdir.py]: shadows builtin `bytes`

Comment (zacharyvoase) [inline clrypt/encdir.py]: Fixed!

File: clrypt/__init__.py (hunk 1 above)
Comment (zacharyvoase) [inline clrypt/__init__.py]: Done :)

**Comment by bhatman** [comment]
this is awesome! LGTM. like the docstrings and tests, and thanks for taking the time to ensure that this is backwards-compatible with our current setup.

in terms of further work, we could add a TODO to use the pubkey signature rather than the key ID that we compute, as you mentioned. 

beyond that, we should think about the feasibility of removing secrets from our codebase completely, but I also think that this approach will continue to work for the time being.

**Comment by zacharyvoase** [comment]
Thanks! I'll merge. OK if I bump to v0.2.0 + tag?

**Comment by bhatman** [comment]
sounds good

**Pull request merged**